### PR TITLE
Next generation building system

### PR DESCRIPTION
**Pull request by sdake**
This PR lints/builds/test all code using pinned tooling out of the
istio/tools repository.

This is part of the build-tools standardization effort.  Some of the
original makefile potentially contains cruft.  I could use some help
deteremining what is cruft and what is necessary for correct operation.

The goal of this work is to standardize our toolchain across all
deliverables of Istio.

### CODE REVIEW

**Comment by sdake** [comment]
the CI image being used for DIND is here: https://github.com/istio/tools/pull/246

**Comment by sdake** [comment]
related work from @geeknoid - we are working together on this and will converge on a solution: https://github.com/istio/pkg/pull/46.  Comments welcome in either PR.

File: Makefile.mk [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	@go test -race ./...
+
+test_with_coverage:
+	@go test -race -coverprofile=coverage.txt -covermode=atomic ./...
+	@curl -s https://codecov.io/bash | bash -s -- -c -F aFlag -f coverage.txt
+
+mandiff:
+	@scripts/run_mandiff.sh
+
+build: vfsgen /work/mesh
```
Comment (morvencao) [inline Makefile.mk]: duplicated with `mesh` target below.

Comment (sdake) [reply Makefile.mk]: changed to `build: mesh`.

**Pull request merged**